deployer.sh

```diff
@@ -129,44 +174,52 @@ sshpass -p${ROOT_PASSWORD} \
                   --inventory=${INVENTORY_PATH} \
                   ${OPENSHIFT_ANSIBLE_PATH}/playbooks/byo/config.yml
 
-SSH_COMMAND="sshpass -p${ROOT_PASSWORD} ssh ${SSH_ARGS} root@${MASTER_HOSTNAME}"
-
 if [ $? -ne '0' ]; then
   RETRCODE=1
-elif [ "${STORAGE_TYPE}" == "external_nfs" ]; then
-      echo "Creating PVs..."
-      sshpass -p${ROOT_PASSWORD} rsync -e "ssh ${SSH_ARGS}" -Pahvz ${TMP_RESOURCE_DIR} root@${MASTER_HOSTNAME}:
-
-      PV_YAML_DIR=`basename ${TMP_RESOURCE_DIR}`
-
-      for PV in `seq -f "vol-%03g.yaml" 1 ${NUM_OF_PVS}`
-      do
-        ${SSH_COMMAND} oc create -f ${PV_YAML_DIR}/${PV}
-      done
-fi
-
-if [ "$INSTALL_MANAGEIQ" == "true" ] && [ "$CONFIGURE_MANAGEIQ_PROVIDER" == "true" ]; then
-  echo "Checking out Ansible 2.4..."
-  git clone https://github.com/ansible/ansible.git
-  pushd ansible
-  git checkout stable-2.4
-  source hacking/env-setup
-  popd
-  ansible --version
-  echo "Configuring OpenShift provider in ManageIQ..."
-
-  export OPENSHIFT_HAWKULAR_ROUTE="$(${SSH_COMMAND}  oc get route --namespace='openshift-infra' -o go-template --template='{{.spec.host}}' hawkular-metrics 2> /dev/null)"
-  export OPENSHIFT_PROMETHEUS_ALERTS_ROUTE="$(${SSH_COMMAND}  oc get route --namespace='openshift-metrics' -o go-template --template='{{.spec.host}}' alerts 2> /dev/null)"
-  export OPENSHIFT_PROMETHEUS_METRICS_ROUTE="$(${SSH_COMMAND}  oc get route --namespace='openshift-metrics' -o go-template --template='{{.spec.host}}' metrics 2> /dev/null)"
-  export OPENSHIFT_CFME_ROUTE="$(${SSH_COMMAND}  oc get route --namespace='openshift-management' -o go-template --template='{{.spec.host}}' httpd 2> /dev/null)"
-  export OPENSHIFT_MASTER_HOST="$(${SSH_COMMAND} oc get nodes -o name |grep master |sed -e 's/nodes\///g')"
-  export OPENSHIFT_CA_CRT="$(${SSH_COMMAND} cat /etc/origin/master/ca.crt)"
-  export OPENSHIFT_MANAGEMENT_ADMIN_TOKEN="$(${SSH_COMMAND} oc sa get-token -n management-infra management-admin)"
-
-  ansible-playbook --extra-vars "provider_name=${NAME_PREFIX} cfme_route=https://${OPENSHIFT_CFME_ROUTE}" ${WORKSPACE}/miqplaybook.yml
-  if [ $? -ne '0' ]; then
-    RETRCODE=1
-  fi
+else
+    if [ "${INSTALL_PROMETHEUS}" == "true" ]; then
+        echo "Creating iSCSI pv (for Prometheus)..."
+        export ISCSI_TARGET_PORTAL
+        export ISCSI_IQN
+        envsubst < "${WORKSPACE}/iscsi-pv-template.yaml" > iscsi_pv.yaml
+        sshpass -p${ROOT_PASSWORD} rsync -e "ssh ${SSH_ARGS}" -Pahvz iscsi_pv.yaml root@${MASTER_HOSTNAME}:
+        ${SSH_COMMAND} oc create -f iscsi_pv.yaml
+    fi
+    if [ "${STORAGE_TYPE}" == "external_nfs" ]; then
+          echo "Creating PVs..."
+          sshpass -p${ROOT_PASSWORD} rsync -e "ssh ${SSH_ARGS}" -Pahvz ${TMP_RESOURCE_DIR} root@${MASTER_HOSTNAME}:
+
+          PV_YAML_DIR=`basename ${TMP_RESOURCE_DIR}`
```

Comment:
subshells?

Reply:
I didn't touch this line, just moved it. It was a subshell before...